typing/ctype.ml

```diff
@@ -2690,13 +2681,35 @@ let rec unify (env:Env.t ref) t1 t2 =
     reset_trace_gadt_instances reset_tracing;
     raise( Unify (Errortrace.diff t1 t2 :: trace) )
 
-and unify2 env t1 t2 =
+and unify2 env t1 t2 = unify2_expand env t1 t1 t2 t2
+
+and unify2_rec env t10 t1 t20 t2 =
+  let t1 = repr t1 and t2 = repr t2 in
+  if unify_eq t1 t2 then () else
+  try match (t1.desc, t2.desc) with
+  | (Tconstr (p1, tl1, a1), Tconstr (p2, tl2, a2)) ->
+      if Path.same p1 p2 && tl1 = [] && tl2 = [] && is_public_type !env p1
+      && not (has_cached_expansion p1 !a1 || has_cached_expansion p2 !a2)
+      then begin
+        update_level_for Unify !env t1.level t2;
+        update_scope_for Unify t1.scope t2;
+        link_type t1 t2
```

Comment:
Is it okay to always link `t1` to `t2` rather than the other way around? I know that `unify2_expand` somtimes changes the order of the types, so I was wondering if we ever wanted to do the same here.

Reply:
Here deliberately I do not keep eventual abbreviations (i.e. link to `t2` rather than than `t20`).
The rationale is that since this type is just a type constructor without parameters, an abbreviation cannot be better.
Since `t1` and `t2` describe exactly the same type, modulo the node id, the direction does not matter.